### PR TITLE
Feature/INTG-266 Migrate BurgerPOS terminal settings to EspressoPOS

### PR DESCRIPTION
**Pull request by jwsandeman**

### CODE REVIEW

File: src/components/SettingsPage/SettingsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                aria-label="settings tabs"
+                className={classes.tabs}
+                id="settingsTabs"
+                onChange={handleTabChange}
+                value={tabIndex}
+              >
+                <Tab id="terminalSettingsTab" label="Terminal Settings" />
+                {/* <Tab id="posSettingsTab" label="POS Settings" /> */}
+              </Tabs>
+              {tabIndex === 0 && (
```
Comment (rossi-mx51) [inline src/components/SettingsPage/SettingsPage.tsx:38]: I think the whole thing here could be refactored to;


```
<SettingsPanel index={tabIndex} subtitle={tabIndex === 0 ? Configure the terminal settings and customise the EFTPOS receipts. :  Configure the POS settings. title={....} value={tabIndex} />
```

File: src/hooks/useLocalStorage.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    const item = getLocalStorage(key);
+    if (typeof item === 'string') {
+      return setState(JSON.parse(item));
+    }
+    return setLocalStorage(key, JSON.stringify(initialValue));
+  }, []);
+
+  useEffect(() => {
+    setLocalStorage(key, JSON.stringify(state));
+    if (onStateChanged) {
```
Comment (rossi-mx51) [inline src/hooks/useLocalStorage.tsx:21]: A little typescript thing that you can do here: 

```
if (onStateChanged) { 
    onStateChanged(state);
}
```

=> 

```
onStateChanged?.(state)
```

**Review by rossi-mx51 — CHANGES_REQUESTED** [review]
Great work @jwsandeman , just a little thing to make the code cleaner.

File: src/components/SettingsPage/SettingsPage.tsx (hunk 1 above)
Comment (jwsandeman) [reply src/components/SettingsPage/SettingsPage.tsx]: Thanks for the suggestion.

File: src/hooks/useLocalStorage.tsx (hunk 2 above)
Comment (jwsandeman) [reply src/hooks/useLocalStorage.tsx]: Oh nice. Changed.

**Review by rossi-mx51 — APPROVED** [review]

**Pull request merged**